src/models/timeaxisheaderview.cpp

```diff
@@ -0,0 +1,109 @@
+#include "timeaxisheaderview.h"
+
+#include <QDebug>
+#include <QPainter>
+#include <QtMath>
+
+#include "eventmodel.h"
+#include "filterandzoomstack.h"
+#include "../util.h"
+
+#include "../../3rdparty/PrefixTickLabels/src/PrefixTickLabels.h"
+
+TimeAxisHeaderView::TimeAxisHeaderView(FilterAndZoomStack* filterAndZoomStack, QWidget* parent)
+    : QHeaderView(Qt::Horizontal, parent)
+    , m_filterAndZoomStack(filterAndZoomStack)
+{
+    setMinimumHeight(3*fontMetrics().height() + s_tickHeight);
+    setStretchLastSection(true);
+
+    auto emitHeaderDataChanged = [this]() { emit headerDataChanged(this->orientation(), EventModel::EventsColumn, EventModel::EventsColumn); };
+    connect(filterAndZoomStack, &FilterAndZoomStack::filterChanged, this, emitHeaderDataChanged);
+    connect(filterAndZoomStack, &FilterAndZoomStack::zoomChanged, this, emitHeaderDataChanged);
+}
+
+void TimeAxisHeaderView::setTimeRange(const Data::TimeRange& timeRange)
+{
+    m_timeRange = timeRange;
+    emit headerDataChanged(orientation(), 1, 1);
+}
+
+void TimeAxisHeaderView::paintSection(QPainter* painter, const QRect& rect, int logicalIndex) const
```

Comment:
could you add a few comments to this function please that at least outline the different aspects of what gets painted where? or alternatively put these sections into different named lambdas to better explain/visualize within the code what gets painted where